ext/standard/var.c

```diff
@@ -1254,6 +1269,7 @@ PHP_FUNCTION(unserialize)
 		gc_check_possible_root(ref);
 	}
 
+cleanup:
 	if (class_hash) {
 		zend_hash_destroy(class_hash);
 		FREE_HASHTABLE(class_hash);
```

Comment:
Does this need something similar to line 1263(original) for the max_depth and possibly cur_depth value? (e.g. while calling unserialize with a max_depth, another __unserialize or __wakeup calls unserialize() with a different max_depth.

- E.g. if the inner unserialize() call returned false due to invalid data, throwing an exception, etc.
- I haven't looked at all of the uses of allowed_classes. There may be more things that need to be restored.
- Also, is this backing up the old cur_depth and max_depth in nested unserialize()?

```c
	/* Reset to previous allowed_classes in case this is a nested call */
	php_var_unserialize_set_allowed_classes(var_hash, prev_class_hash);
```